provisioning/miniprov/hare_mp/main.py

```diff
@@ -287,11 +287,23 @@ def save(filename: str, contents: str) -> None:
         f.write(contents)
 
 
+def generate_config(path_to_cdf: str):
+    conf_dir='/var/lib/hare'
+    os.environ['PATH'] += os.pathsep + '/opt/seagate/cortx/hare/bin/'
+    process = subprocess.run(['cfgen', '-o', conf_dir, path_to_cdf],
+                             check=True,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE,
+                             shell=False)
+    return process.returncode
+
+
 def config(args):
     try:
         url = args.config[0]
         filename = args.file[0] or '/var/lib/hare/cluster.yaml'
         save(filename, generate_cdf(url))
+        generate_config(filename)
```

Comment:
Return code is not used. In case of cfgen failure provisioner must error out.

Reply:
That's why I prefer exceptions to error codes :-)

Reply:
Yes, updated the code. Calling `execute() ` function present in this file, in `generate_config()`, which raises exception for failure.